### ISSUE TITLE
Update to latest available version and keep the range.

### ISSUE THREAD

**Issue by voroninp** [NuGet/Home#13256]
### NuGet Product(s) Involved

Visual Studio Package Management UI, dotnet.exe

### The Elevator Pitch

When authoring NuGet packages I prefer to use version ranges: `[x.y.z, x+1)`.
However, every time I update packages with UI the range gets replaced with just version: `[x.y.z, x+1) -> x2.y2.z2`
I'd like to keep the semantics, when updating I'd be able to explicitly tell Nuget to specify the range : `[x.y.z, x+1) -> [x2.y2.z2, x2+1)`.

### Additional Context and Details
Automation of the package updates seems to require a lot.
The output of `dotnet list package --format json` requires manual processing of version ranges:
```json
 {
    "id": "Google.Protobuf",
    "requestedVersion": "[3.25.3, 4)",
    "resolvedVersion": "3.25.3"
}
```

**Comment by martinrrm** [comment]
Hi! Ranges are available in the UI for some project types (Package Reference) you can type it in the Package Manager UI and let you install the package, the range will also be written in the project file (.csproj) 

![image](https://github.com/NuGet/Home/assets/43253759/ac843a45-32f5-45df-a5c6-127505d87787)
![image](https://github.com/NuGet/Home/assets/43253759/830506d5-c792-4d79-991a-49ef8792523e)

**Comment by voroninp** [comment]
Yes, I know, the problem is that you need to remember about the ranges and set them explicitly.
If you just select multiple packages and press `Update` button, this won't work. I'd rather have a sort of update policy per package.

**Comment by CharlieWallace17** [comment]
Is there any intention to have this prioritized in the near future? Our team often runs into issues similar to the above due to package versions being automatically overwritten
